lib/reek/configuration/schema_validator.rb

```diff
@@ -1,38 +1,37 @@
 # frozen_string_literal: true
 
-require 'yaml'
-require_relative '../cli/silencer'
-Reek::CLI::Silencer.without_warnings { require 'kwalify' }
 require_relative '../errors/config_file_error'
+require_relative 'schema'
 
 module Reek
   module Configuration
     #
     # Schema validator module.
     #
     class SchemaValidator
-      SCHEMA_FILE_PATH = File.expand_path('./schema.yml', __dir__)
-
       def initialize(configuration)
         @configuration = configuration
-        @validator = CLI::Silencer.without_warnings do
-          schema_file = Kwalify::Yaml.load_file(SCHEMA_FILE_PATH)
-          Kwalify::Validator.new(schema_file)
-        end
+        config_directories = configuration['directories']&.keys || []
+        @validator = Reek::Configuration::Schema.schema(config_directories)
       end
 
       def validate
-        errors = CLI::Silencer.without_warnings { @validator.validate @configuration }
-        return if !errors || errors.empty?
+        result = @validator.call(@configuration)
+        return if result.success?
 
-        raise Errors::ConfigFileError, error_message(errors)
+        raise Errors::ConfigFileError, error_message(result.errors)
+      rescue NoMethodError
+        raise Errors::ConfigFileError, 'unrecognized configuration data'
```

Comment:
Raise shorter error messages, because they get caught and used in lib/reek/configuration/configuration_file_finder.rb